src/infra/flow-system/postgres/src/postgres_flow_event_store.rs

```diff
@@ -592,12 +592,21 @@ impl FlowEventStore for PostgresFlowEventStore {
 
             let mut query_stream = sqlx::query!(
                 r#"
-                SELECT flow_id FROM flows
-                    WHERE dataset_id = $1
-                    AND (cast($2 as dataset_flow_type) IS NULL OR dataset_flow_type = $2)
-                    AND (cast($3 as flow_status_type) IS NULL OR flow_status = $3)
-                    AND (cast($4 as TEXT[]) IS NULL OR initiator = ANY($4))
-                ORDER BY flow_id DESC
+                WITH unsorted_flows AS
+                    (SELECT
+                        f.flow_id,
+                        f.flow_status,
+                        MAX(e.event_time) as last_event_time
```

Comment:
Flow already keeps `last_event_id`, which points on the last occurred event. It was an important denormalization. As event identifiers are a sequence, you can always replace time comparison on id comparison - the higher is the id of the event, the later it was.

So, the query could be simplified signifficantly, as JOIN with `flow_events` table can be avoided..